Sources/Redis/Database/RedisDatabase.swift

```diff
@@ -13,14 +13,23 @@ public final class RedisDatabase: Database {
 
     /// See `Database`.
     public func newConnection(on worker: Worker) -> EventLoopFuture<RedisClient> {
-        return RedisClient.connect(
+        let connect = RedisClient.connect(
             hostname: config.hostname,
             port: config.port,
             password: config.password,
             on: worker
         ) { error in
             print("[Redis] \(error)")
         }
+
+        guard let database = config.database else {
+            return connect
+        }
+
+        return connect.flatMap { client in
```

Comment:
Any reason you're doing the selecting of the database here and not in the `connect` method?

Reply:
Right, in my implementation I just passed the password into it. But this works for me!